forem/forem
# [Team Email Login] Admin enable/disable recaptcha during email signup

## What type of PR is this? (check all applicable)


## Description
***Update: I incorporated @lisasy's UX changes, and other requested QA changes to the PR**
This PR grants Forem admins the ability to require reCAPTCHA during email signup. The PR also implements the reCAPTCHA check during email signup.

## Related Tickets & Documents
None

## QA Instructions *(updated)*
1. Sign into your local app as an `admin` (or `super_admin`) with `single_resource_admin` role (important), navigate to `/admin/config` and toggle the Authentication section open.
2. Check the `Enable Google reCAPTCHA for email password registration` option, which should reveal the fields for the Google reCAPTCHA keys. Enter the keys (see guide right after this paragraph for how to obtain reCAPTCHA keys for `localhost`). Be sure to `Update Site Configuration` at the bottom of the section after making changes!
2. In an incognito window, navigate to `localhost:<port>/users/sign_up?state=email_signup` and test the signup with reCAPTCHA completed. You should be able to create an account successfully.
3. Run through the signup process again, but do not complete the reCAPTCHA. You should be blocked from registering a new account.



**Getting Google ReCaptcha Keys:**
1. Google "google reCAPTCHA admin" and navigate to your admin console
2. Click the "+" in the top right area to add keys for a new site
3. Enter any label
4. Choose `reCAPTCHA v2`
5. Add TWO domains - `example.com` (or something else random) and `localhost`. The first random domain prevents 
 errors being thrown on your local app, as technically Google does not support `localhost` as the sole domain.
6. Accept Terms and Submit


## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?
None

## [optional] What gif best describes this PR or how it makes you feel?

![woman blocked by guards](https://media.giphy.com/media/1kTI9UeEWBlz10qR8Y/giphy-downsized.gif)
